aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -77,8 +77,13 @@ compute_result_type(at::ArrayRef<OperandInfo> operands, const F& predicate) {
   ScalarType dtype = ScalarType::Undefined;
   for (auto& op : operands) {
     if (!op.tensor.defined()) continue;
-    if (!predicate(op.tensor)) continue;
-    auto tensor_dtype = op.tensor.scalar_type();
+    if (!predicate(op)) continue;
+    ScalarType tensor_dtype;
+    if (op.tensor.unsafeGetTensorImpl()->is_wrapped_number() && isFloatingType(op.tensor.scalar_type())) {
```

Comment:
right, there might be other unrelated issues to look into here too (e.g. the boolean issue we discussed), but in the existing code in ScalarOps.h we always use double for floating point and long for integral types.

Reply:
This looks correct to me. This case is described in the issue https://github.com/pytorch/pytorch/issues/9515, specifically in infer_scalar_type.

@nairbv is right, we always wrap Python numbers as int64 or double tensors  to avoid losing precision.

Reply:
sounds good, would be nice to document that.